### PR TITLE
feat(framework) Add deployment engine executor

### PR DESCRIPTION
**Pull request by charlesbvll**
### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: src/py/flwr/superexec/deployment_plugin.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        assert self.stub is not None
+
+        req = CreateRunRequest(fab_id=fab_id, fab_version=fab_version)
+        res = self.stub.CreateRun(request=req)
+        return int(res.run_id)
+
+    def _install_fab(self, fab_file: bytes) -> Path:
+        return install_from_fab(fab_file, None, True)
+
+    def start_run(self, fab_file: bytes, ttl: Optional[float] = None) -> RunTracker:
```
Comment (jafermarq) [inline src/py/flwr/superexec/deployment_plugin.py:52]: Do we want to pass the `ttl` here ? or should it be relegated to a future `ServerApp` & `ClientApp` config?

File: src/py/flwr/superexec/deployment_plugin.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+"""Deployment engine executor plugin."""
```
Comment (jafermarq) [inline src/py/flwr/superexec/deployment_plugin.py:1]: needs copyright notice 😇

File: src/py/flwr/superexec/deployment_plugin.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            self._connect()
+
+        assert self.stub is not None
+
+        req = CreateRunRequest(fab_id=fab_id, fab_version=fab_version)
+        res = self.stub.CreateRun(request=req)
+        return int(res.run_id)
+
+    def _install_fab(self, fab_file: bytes) -> Path:
+        return install_from_fab(fab_file, None, True)
```
Comment (jafermarq) [inline src/py/flwr/superexec/deployment_plugin.py:50]: v minor. Why not calling `install_from_fab()` directly in the `start_run()` method?

Comment (charlesbvll) [reply src/py/flwr/superexec/deployment_plugin.py:50]: Good point, I'll change that

File: src/py/flwr/superexec/deployment_plugin.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            )
+
+            return RunTracker(
+                run_id=run_id,
+                proc=subprocess.Popen(
+                    [
+                        "flower-server-app",
+                        "build_demo.server:app",
```
Comment (danieljanes) [inline src/py/flwr/superexec/deployment_plugin.py:87]: This needs to be replaced

**Review by danieljanes — APPROVED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**